sphinx/testing/fixtures.py

```diff
@@ -223,11 +222,15 @@ def if_graphviz_found(app):
 
 
 @pytest.fixture(scope='session')
-def sphinx_test_tempdir():
+def sphinx_test_tempdir(tmpdir_factory):
     """
     temporary directory that wrapped with `path` class.
     """
-    return util.path(os.environ.get('SPHINX_TEST_TEMPDIR', gettempdir())).abspath()
+    tmpdir = os.environ.get('SPHINX_TEST_TEMPDIR')
```

Comment:
I think a deprecation declaration (or comment) is needed if this `SPHINX_TEST_TEMPDIR` is kept for backward compatibility,